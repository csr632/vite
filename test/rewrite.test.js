const {
  transformCjsImport
} = require('../dist/node/server/serverPluginModuleRewrite')

describe('rewrite commonjs imports', () => {
  const packageName = 'react'
  const resolvedPackageName = '/@modules/react/index.js'
  const cases = [
    `import { useState } from 'react'`,
    `import { useState, useEffect } from 'react'`,
    `import * as React from 'react'`,
    `import { useState as something } from 'react'`,
    `import { useState as something, useEffect } from 'react'`,
    `import { useState as something, useEffect as alias } from 'react'`,
    `import { default as Default } from 'react'`,
    `import { default as Default, useEffect } from 'react'`,
    `import React from 'react'`,
    `import React, { useState } from 'react'`
  ]
  for (let testCase of cases) {
    test(`"${testCase}"`, () => {
      const res = transformCjsImport(
        testCase,
        packageName,
        resolvedPackageName,
        0
      )
      expect(res).toMatchSnapshot()
    })
  }
})
