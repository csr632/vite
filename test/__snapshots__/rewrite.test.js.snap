// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rewrite commonjs imports "import * as React from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"/@modules/react/index.js\\";
const React = $viteCjsImport0_react;"
`;

exports[`rewrite commonjs imports "import { default as Default } from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"/@modules/react/index.js\\";
const Default = $viteCjsImport0_react;"
`;

exports[`rewrite commonjs imports "import { default as Default, useEffect } from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"/@modules/react/index.js\\";
const Default = $viteCjsImport0_react;
const useEffect = $viteCjsImport0_react[\\"useEffect\\"];"
`;

exports[`rewrite commonjs imports "import { useState } from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"/@modules/react/index.js\\";
const useState = $viteCjsImport0_react[\\"useState\\"];"
`;

exports[`rewrite commonjs imports "import { useState as something } from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"/@modules/react/index.js\\";
const something = $viteCjsImport0_react[\\"useState\\"];"
`;

exports[`rewrite commonjs imports "import { useState as something, useEffect } from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"/@modules/react/index.js\\";
const something = $viteCjsImport0_react[\\"useState\\"];
const useEffect = $viteCjsImport0_react[\\"useEffect\\"];"
`;

exports[`rewrite commonjs imports "import { useState as something, useEffect as alias } from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"/@modules/react/index.js\\";
const something = $viteCjsImport0_react[\\"useState\\"];
const alias = $viteCjsImport0_react[\\"useEffect\\"];"
`;

exports[`rewrite commonjs imports "import { useState, useEffect } from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"/@modules/react/index.js\\";
const useState = $viteCjsImport0_react[\\"useState\\"];
const useEffect = $viteCjsImport0_react[\\"useEffect\\"];"
`;

exports[`rewrite commonjs imports "import React from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"/@modules/react/index.js\\";
const React = $viteCjsImport0_react;"
`;

exports[`rewrite commonjs imports "import React, { useState } from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"/@modules/react/index.js\\";
const React = $viteCjsImport0_react;
const useState = $viteCjsImport0_react[\\"useState\\"];"
`;
